src/hazelcore/HTyp.re

```diff
@@ -118,14 +124,14 @@ let rec complete =
 
 let rec join = (j, ty1, ty2) =>
   switch (ty1, ty2) {
-  | (_, Hole) =>
+  | (_, Unknown(x)) =>
     switch (j) {
-    | GLB => Some(Hole)
+    | GLB => Some(Unknown(x))
```

Comment:
I think the provenance here should always be Internal rather than inheriting from the one type, since there are potentially two Unknowns involved and this would just be picking one of them arbitrarily.